lib/osml/osml_http_endpoint.ts

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright 2023 Amazon.com, Inc. or its affiliates.
+ */
+
+import { Duration, RemovalPolicy } from "aws-cdk-lib";
+import { IVpc } from "aws-cdk-lib/aws-ec2";
+import {
+  Cluster,
+  Compatibility,
+  ContainerImage, LogDriver, Protocol,
+  TaskDefinition
+} from "aws-cdk-lib/aws-ecs";
+import { ApplicationLoadBalancedFargateService } from "aws-cdk-lib/aws-ecs-patterns";
+import { IRole } from "aws-cdk-lib/aws-iam";
+import { Construct } from "constructs";
+import { OSMLAccount } from "./osml_account";
+import { LogGroup, RetentionDays } from "aws-cdk-lib/aws-logs";
+
+export interface OSMLHTTPEndpointProps {
+  account: OSMLAccount;
+  vpc: IVpc;
+  image: ContainerImage;
+  clusterName: string;
+  role: IRole;
+  memory: number;
+  cpu: number;
+  hostPort: number;
+  containerPort: number;
+  healthcheckPath: string;
+  loadBalancerName: string;
+  containerEnv?: {};
+}
+
+export class OSMLHTTPModelEndpoint extends Construct {
+  public networkHTTPEndpoint: ApplicationLoadBalancedFargateService;
+  public logGroup: LogGroup;
+  public removalPolicy: RemovalPolicy;
+  constructor(scope: Construct, id: string, props: OSMLHTTPEndpointProps) {
+    super(scope, id);
+    const httpEndpointCluster = new Cluster(this, props.clusterName, {
+      clusterName: props.clusterName,
+      vpc: props.vpc
+    });
+    // setup a removal policy
+    this.removalPolicy = props.account.prodLike
+      ? RemovalPolicy.RETAIN
+      : RemovalPolicy.DESTROY;
+
+    // log group for MR container
+    this.logGroup = new LogGroup(this, "MRServiceLogGroup", {
+      logGroupName: "/aws/OSML/HTTPEndpoint",
+      retention: RetentionDays.TEN_YEARS,
+      removalPolicy: this.removalPolicy
+    });
+
+    const taskDefinition = new TaskDefinition(
+      this,
+      "HTTPEndpointFargateTaskDefinition",
+      {
+        memoryMiB: props.memory.toString(),
+        cpu: props.cpu.toString(),
+        compatibility: Compatibility.FARGATE,
+        taskRole: props.role,
+        ephemeralStorageGiB: 100
+      }
+    );
+    taskDefinition.addContainer("HTTPEndpointContainer", {
+      image: props.image,
+      portMappings: [
+        {
+          containerPort: props.containerPort,
+          hostPort: props.hostPort,
+          protocol: Protocol.TCP,
+        }
+      ],
+      logging: LogDriver.awsLogs({
+        logGroup: this.logGroup,
+        streamPrefix: "OSML"
+      }),
+      environment: props.containerEnv
+    });
+
+    this.networkHTTPEndpoint = new ApplicationLoadBalancedFargateService(
```

Comment:
Doesn't this setup a load balancer that is internet facing by default (see: https://docs.aws.amazon.com/cdk/api/v2/python/aws_cdk.aws_ecs_patterns/ApplicationLoadBalancedFargateService.html   ...  public_load_balancer (Optional[bool]) – Determines whether the Load Balancer will be internet-facing. Default: true)?

If so how is the endpoint protected?

Reply:
I have updated the deployment to not expose the load balancer - great catch.